### PR TITLE
ADEN-10890 Send RealVu detection result with bidders calls

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: src/ad-bidders/prebid/prebid-adapter.ts [hunk 1]
```diff
@@ -39,12 +40,21 @@ export abstract class PrebidAdapter {
...
 			src: [context.get('src') || ''],
-			pos: [slotName],
+			pos: [placementName],
 			...customTargeting,
 		};
+
+		const slotName = getSlotNameByBidderAlias(placementName);
+		const realVu = context.get(`slots.${slotName}.targeting.realvu`);
+
+		if (realVu) {
```
Comment (preludek) [inline src/ad-bidders/prebid/prebid-adapter.ts:54]: Can realVu be an empty object?

File: src/ad-engine/utils/logger.ts [hunk 2]
```diff
@@ -36,11 +36,11 @@ export function warner(warnGroup: string, ...warnValues: any[]): void {
 
 export function communicator(
 	communicatorEvent: string,
-	communicatorValues: CommunicatorPayload,
+	communicatorValues: CommunicatorPayload | {} = {},
```
Comment (preludek) [inline src/ad-engine/utils/logger.ts:39]: Why this change?

Comment (dmnsobczak) [reply src/ad-engine/utils/logger.ts:39]: I added this cool feature to emit communicator events without payload in previous code iteration and I decided to leave it, because it will be useful in the future. Oh wait, the future is now! It is also part of Tomek's PR: https://github.com/Wikia/ad-engine/pull/1026/files#diff-50e4c1249aa3095f147eea2fd8ee626809020070724d90d931b08ce57724467cR39

File: src/ad-bidders/prebid/prebid-adapter.ts (hunk 1 above)
Comment (dmnsobczak) [reply src/ad-bidders/prebid/prebid-adapter.ts:54]: `context.get` will return `undefined` if this key is not set and it will be if RealVu will be disabled. Prebid sets it here: https://github.com/Wikia/ad-engine/pull/1006/files#diff-bfd0be70e9f82c032464ba814d31af004445e48bbfd4cf4becdd0da1b1c83390R180

File: src/ad-bidders/prebid/prebid-adapter.ts [hunk 3]
```diff
@@ -39,12 +40,21 @@ export abstract class PrebidAdapter {
 		);
 	}
 
-	protected getTargeting(slotName, customTargeting = {}): Dictionary {
-		return {
+	protected getTargeting(placementName, customTargeting = {}): Dictionary {
```
[the remark this answers is not in the document]
Comment (dmnsobczak) [reply src/ad-bidders/prebid/prebid-adapter.ts:43]: Done!

File: src/ad-bidders/prebid/index.ts [hunk 4]
```diff
@@ -142,22 +142,57 @@ export class PrebidProvider extends BidderProvider {
...
+	}
+
 	async applyConfig(config: Dictionary): Promise<void> {
 		const pbjs: Pbjs = await pbjsFactory.init();
 
 		return pbjs.setConfig(config);
 	}
 
+	async applyAnalytics(): Promise<void> {
+		const pbjs: Pbjs = await pbjsFactory.init();
```
[the remark this answers is not in the document]
Comment (dmnsobczak) [reply src/ad-bidders/prebid/index.ts:160]: You're right. Removed.
It's a copy-paste convention I guess. We can agree to skip redundant types.

**Review by preludek — APPROVED** [review]

File: spec/ad-bidders/prebid/adapters/appnexus-ast.spec.ts [hunk 5]
```diff
@@ -90,4 +90,51 @@ describe('AppnexusAst bidder adapter', () => {
...
+								skippable: false,
+								playback_method: ['auto_play_sound_off'],
+							},
+						},
+					},
+				],
+			},
+		]);
+
+		context.set('bidders.prebid.additionalKeyvals.appnexus', false);
```
[the remark this answers is not in the document]
Comment (dmnsobczak) [reply spec/ad-bidders/prebid/adapters/appnexus-ast.spec.ts:137]: It's basically wrong that we need to reset it or modified context will interfere on other unit test :<
Moved!
